src/http/cookie.cr

```diff
@@ -24,27 +24,49 @@ module HTTP
     property http_only : Bool
     property samesite : SameSite?
     property extension : String?
+    property max_age : Time::Span?
+    getter creation_time : Time
 
     def_equals_and_hash name, value, path, expires, domain, secure, http_only
 
-    def initialize(@name : String, value : String, @path : String = "/",
+    @[Deprecated("Use named arguments instead.")]
+    def self.new(_name : String, _value : String, _path : String = "/",
+                 _expires : Time? = nil, _domain : String? = nil,
+                 _secure : Bool = false, _http_only : Bool = false,
+                 _samesite : SameSite? = nil, _extension : String? = nil) : self
+      new(
+        _name, _value,
+        path: _path, expires: _expires, domain: _domain, secure: _secure,
+        http_only: _http_only, samesite: _samesite, extension: _extension
+      )
+    end
+
+    def self.new(_name : String, _value : String) : self
+      new(name: _name, value: _value)
+    end
```

Comment:
Why's this initializer needed?

Reply:
Otherwise `Cookie.new(name, value)` would be a deprecation warning because the deprecated `.new` method shadows the implicit `.new` from `initialize`. But this basic constructor should continue to work, only the optional config arguments should be named-only.

Reply:
So why not use `def self.new(name : String, value : String) : self` signature?

Reply:
The named arguments must not match the ones it delegates to or it would create an infinite recursion.

Reply:
Ah, of course, thanks for bearing with me :)